.github/workflows/pull_request_verify.yaml

```diff
@@ -0,0 +1,191 @@
+name: Pull Requests
+
+# When someone comment "/deploy" on anyone PR,
+# this workflow will be triggered and run.
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened, labeled ]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+  CKB_CLI_VERSION: v0.40.0  
+jobs:
+#job for checkout the ckb
+  checkout: 
+    name: ckb-checkout  
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+        -  Linux-runner
+        -  mac-runner
+    steps:
+    - name: Set env
+      run: echo "CARGO_TARGET_DIR=$(echo $HOME/target)" >> $GITHUB_ENV
+    - uses: actions/checkout@v2
+      with:
+        submodules: false
+        fetch-depth: 2
+        ref: ${{ github.event.pull_request.head.sha }}
+        repository: ${{github.event.pull_request.head.repo.full_name}}
+        #cargo build 的target，试一下在job 层加 cargo build targe
+    - run: |
+        echo "runner os is ${{ runner.os }}"
+        if [ ${{ runner.os }} == "macOS" ]; then
+        export PATH="/Users/administrator/.cargo/bin:$PATH"
+        fi
+        echo $PATH
+        make build          
+  Test: 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+        -  Linux-runner
+        -  mac-runner
+    needs: checkout
+    steps:
+    - name: Set env
+      run: echo "CARGO_TARGET_DIR=$(echo $HOME/target)" >> $GITHUB_ENV
+    - name: Test
+      run: |
+        if [ ${{ runner.os }} == "macOS" ]; then
+        export PATH="/Users/administrator/.cargo/bin:$PATH"
+        fi
+        make test
+        # rm -rf $HOME/.cargo/registry
+  Linters: 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+        -  Linux-runner
+        -  mac-runner
+    needs: Test
+    steps:
+    - name: Set env
+      run: echo "CARGO_TARGET_DIR=$(echo $HOME/target)" >> $GITHUB_ENV
+    - name: Linters
+      run: |
+         if [ ${{ runner.os }} == "macOS" ]; then
+         export PATH="/Users/administrator/.cargo/bin:$PATH"
+         fi
+         cargo fmt --version || travis_retry rustup component add rustfmt
+         cargo clippy --version || travis_retry rustup component add clippy
+         make fmt
+         make clippy
+         git diff --exit-code Cargo.lock
+  Latest_Linters:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+        -  Linux-runner
+        -  mac-runner
+    needs: Linters 
+    continue-on-error: true
+    steps:
+    - name: Set env
+      run: echo "CARGO_TARGET_DIR=$(echo $HOME/target)" >> $GITHUB_ENV
+    - name: Latest_Linters
+      run: |
+        if [ ${{ runner.os }} == "macOS" ]; then
+        export PATH="/Users/administrator/.cargo/bin:$PATH"
+        rustup component add rustfmt-preview
```

Comment:
I would like to suggest that move these pre-install toolchains into a single step or a single job, and other jobs dependend this pre-install job.

Reply:
add the pre step to install rustup component